### PR TITLE
Query existing Tables 

### PR DESCRIPTION
**Pull request by scopatz**
Yet another QueryableBackend API improvement for querying what tables actually exist. Depends on #1062.

### CODE REVIEW

**Comment by scopatz** [comment]
rebased.

**Comment by scopatz** [comment]
Ping!

**Comment by scopatz** [comment]
Ping again!

**Comment by gidden** [comment]
This looks good to me. There is some funny rebase business going on. @rwcarlsen, any ideas?

**Comment by scopatz** [comment]
Rebased.

**Comment by rwcarlsen** [comment]
rebase? - because I know it is your favorite git command.

**Comment by scopatz** [comment]
Rebased. And slight correction, rebase is my favorite subversion command :)

File: src/sqlite_back.cc [hunk 1]
```diff
@@ -238,6 +238,20 @@ std::map<std::string, DbTypes> SqliteBack::ColumnTypes(std::string table) {
...
 
+std::set<std::string> SqliteBack::Tables() {
+  using std::set;
+  using std::string;
+  set<string> rtn;
+  std::string sql = "SELECT name FROM sqlite_master WHERE type='table';";
+  SqlStatement::Ptr stmt;
+  stmt = db_.Prepare(sql);
+  int i;
+  for (i = 0; stmt->Step(); ++i) {
```
Comment (rwcarlsen) [inline src/sqlite_back.cc]: `while (stmt->Step())`

Comment (scopatz) [inline src/sqlite_back.cc]: I did exactly what is used in the constructor: https://github.com/cyclus/cyclus/blob/develop/src/sqlite_back.cc#L45 Is there are reason to not follow this pattern?

Comment (rwcarlsen) [inline src/sqlite_back.cc]: You should only copy me when I'm writing good code :-)
`https://github.com/cyclus/cyclus/blob/develop/src/sqlite_back.cc#L61`

Comment (scopatz) [inline src/sqlite_back.cc]: I see how it is ;).  Change incoming.

**Comment by scopatz** [comment]
Should be ready now.

**Comment by rwcarlsen** [comment]
thanks @scopatz - especially for being patient with my sluggishness and nitpickiness.  And also thanks for the dev-list discussion - it was very helpful to me.  I'll merge as soon as CI passes.

**Comment by scopatz** [comment]
CI passes. No worries.  I hope it is worth it.

**Pull request merged**